geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/Message.java

```diff
@@ -882,6 +1000,12 @@ void readPayloadFields(final int numParts, final int len) throws IOException {
     int readSecurePart = checkAndSetSecurityPart();
 
     int bytesRemaining = len;
+
+    if (this.ioFilter != null) {
```

Comment:
In some cases we have a filter even if not using SSL, NioPlainEngine. What is the difference between ioFilter being null and it being an instance of NioPlainEngine? Would it be possible to have it never be null? If so, it would be nice to get rid of all these null checks

Reply:
I will investigate, but I think there are cases when for socketchannel there are not defined iofilter.